### PR TITLE
Change default blas_info dictionary in cmodule

### LINKED ISSUE #441: BUG: using numpy code that has been deprecated in v1.26.0
### Describe the issue:

Numpy v1.26.0 was released on Sept. 16, 2023. It no longer has a `get_info()` method inside numpy.__config__ which is used https://github.com/pymc-devs/pytensor/blob/main/pytensor/link/c/cmodule.py#L2720. This causes package import to fail because pytensor doesn't have a upper limit on the version of numpy that can be loaded.

### Reproducable code example:

```python
# with numpy 1.26.0
import pytensor
```

### Error message:

```shell
import pytensor
/opt/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/pytensor/__init__.py:119: in <module>
    from pytensor import scalar, tensor
/opt/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/pytensor/tensor/__init__.py:106: in <module>
    from pytensor.tensor import (  # noqa
/opt/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/pytensor/tensor/blas.py:102: in <module>
    from pytensor.tensor.blas_headers import blas_header_text, blas_header_version
/opt/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/pytensor/tensor/blas_headers.py:1015: in <module>
    if not config.blas__ldflags:
/opt/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/pytensor/configparser.py:321: in __get__
    val_str = self.default()
/opt/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/pytensor/link/c/cmodule.py:2720: in default_blas_ldflags
    blas_info = np.__config__.get_info("blas_opt")
E   AttributeError: module 'numpy.__config__' has no attribute 'get_info'
Error: Process completed with exit code 4.
```

### PyTensor version information:

pytensor v2.16.1

### Context for the issue:

_No response_

### LINKED ISSUE #443: BUG: Numpy New Release incompatible with Pytensor `blas_headers`
### Describe the issue:

Two days ago Numpy released a new version `1.26.0` [documentation here](https://numpy.org/news/). 

The new version doesn't have anymore the attribute `get_info()`, which is used in the file `cmodule.py` ([link here](https://github.com/pymc-devs/pytensor/blob/main/pytensor/link/c/cmodule.py)) on the line 2720.

Consequently, if a user tries to import `pymc` with `Numpy > 1.25.0`, he will get an error, given that these modules are used on the package.

### Reproducable code example:

```python
%pip install pytensor
%pip install numpy==1.26.0
from pytensor.tensor.blas_headers import blas_header_text, blas_header_version
```

```python
import numpy as np
np.__config__.get_info("blas_opt")
```

### Error message:

```shell
Traceback (most recent call last):
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-7295c231-b117-4c44-8446-41f1912bae58/lib/python3.9/site-packages/pytensor/configparser.py", line 201, in fetch_val_for_key
    return self._pytensor_cfg.get(section, option)
  File "/usr/lib/python3.9/configparser.py", line 781, in get
    d = self._unify_values(section, vars)
  File "/usr/lib/python3.9/configparser.py", line 1149, in _unify_values
    raise NoSectionError(section) from None
configparser.NoSectionError: No section: 'blas'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-7295c231-b117-4c44-8446-41f1912bae58/lib/python3.9/site-packages/pytensor/configparser.py", line 317, in __get__
    val_str = cls.fetch_val_for_key(self.name, delete_key=delete_key)
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-7295c231-b117-4c44-8446-41f1912bae58/lib/python3.9/site-packages/pytensor/configparser.py", line 205, in fetch_val_for_key
    raise KeyError(key)
KeyError: 'blas__ldflags'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/databricks/python/lib/python3.9/site-packages/IPython/core/interactiveshell.py", line 3378, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<command-151862714626631>", line 2, in <module>
    from pytensor.tensor.blas_headers import blas_header_text, blas_header_version
  File "/databricks/python_shell/dbruntime/PythonPackageImportsInstrumentation/__init__.py", line 171, in import_patch
    original_result = python_builtin_import(name, globals, locals, fromlist, level)
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-7295c231-b117-4c44-8446-41f1912bae58/lib/python3.9/site-packages/pytensor/__init__.py", line 119, in <module>
    from pytensor import scalar, tensor
  File "/databricks/python_shell/dbruntime/PythonPackageImportsInstrumentation/__init__.py", line 171, in import_patch
    original_result = python_builtin_import(name, globals, locals, fromlist, level)
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-7295c231-b117-4c44-8446-41f1912bae58/lib/python3.9/site-packages/pytensor/tensor/__init__.py", line 106, in <module>
    from pytensor.tensor import (  # noqa
  File "/databricks/python_shell/dbruntime/PythonPackageImportsInstrumentation/__init__.py", line 171, in import_patch
    original_result = python_builtin_import(name, globals, locals, fromlist, level)
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-7295c231-b117-4c44-8446-41f1912bae58/lib/python3.9/site-packages/pytensor/tensor/blas.py", line 102, in <module>
    from pytensor.tensor.blas_headers import blas_header_text, blas_header_version
  File "/databricks/python_shell/dbruntime/PythonPackageImportsInstrumentation/__init__.py", line 171, in import_patch
    original_result = python_builtin_import(name, globals, locals, fromlist, level)
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-7295c231-b117-4c44-8446-41f1912bae58/lib/python3.9/site-packages/pytensor/tensor/blas_headers.py", line 1015, in <module>
    if not config.blas__ldflags:
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-7295c231-b117-4c44-8446-41f1912bae58/lib/python3.9/site-packages/pytensor/configparser.py", line 321, in __get__
    val_str = self.default()
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-7295c231-b117-4c44-8446-41f1912bae58/lib/python3.9/site-packages/pytensor/link/c/cmodule.py", line 2720, in default_blas_ldflags
    blas_info = np.__config__.get_info("blas_opt")
AttributeError: module 'numpy.__config__' has no attribute 'get_info'
```

### PyTensor version information:

Python 3.9.5
Pytensor version (2.16.1)

### Context for the issue:

_No response_

### PR DESCRIPTION
**Pull request by lucianopaz**
Closes #441 

### Motivation for these changes

`numpy.__config__.get_info` has been deprecated and removed, so we have to set default link libraries some other way.

### Implementation details

This is a very rough first implementation. The idea is to use something like python's `sys.path` to look for libraries files (this would match a conda env lib folder and also the system level python lib folders, I still need to figure out what to do with venvs). Then look for some default libraries to link against, like mkl or openblas. I added LAPACK, but maybe the c code that is generated by `pytensor` doesn't have the proper includes to be able to link against that.

I haven't changed any other code yet, but the long term goal would be to remove most of the blas ldflag logic from the startup. In particular, I flagged a couple of platforms for deprecation (EPD and Canopy, which I think are no longer developed anyway)

### Checklist
+ [X] Explain motivation and implementation 👆
+ [X] Make sure that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html).
+ [X] Link relevant issues, preferably in [nice commit messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+ [X] The commits correspond to [_relevant logical changes_](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes). Note that if they don't, we will [rewrite/rebase/squash the git history](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#_rewriting_history) before merging.
+ [x] Are the changes covered by tests and docstrings?
+ [X] Fill out the short summary sections 👇

## Major / Breaking Changes
- `default_blas_ldflags` has changed and might give different default link flags. It no longer looks at `numpy` for blas information, but rather searches some common system paths to try to find BLAS libraries to link against. The order of priorities is 1) MKL, 2) OPENBLAS, 3) LAPACK, 4) BLAS

## New features
- ...

## Bugfixes
- ...

## Documentation
- ...

## Maintenance
- ...

### CODE REVIEW

File: pytensor/link/c/cmodule.py [hunk 1]
```diff
@@ -2716,11 +2717,77 @@ def default_blas_ldflags():
...
+                # First try to find mkl
+                found_libs = check_required_file(
+                    all_libs,
+                    required_regexs=["mkl_core", "mkl_rt", "mkl_(gnu|intel)_thread"],
+                )
+                return found_libs
+            except RuntimeError:
+                pass
+            try:
+                # Second try to find openblas
```
Comment (twiecki) [inline pytensor/link/c/cmodule.py:2749]: What about accelerate?

Comment (lucianopaz) [reply pytensor/link/c/cmodule.py:2768]: That wasn't ever used as a default. I don't know how to test it. I'm not even sure how to test whether LAPACK works.

Comment (twiecki) [reply pytensor/link/c/cmodule.py:2767]: Do we even link against LAPACK in the c-backend? I couldn't find much reference. https://github.com/search?q=repo%3Apymc-devs%2Fpytensor%20lapack&type=code

Comment (twiecki) [reply pytensor/link/c/cmodule.py:2767]: Just talked to @aseyboldt, who said that yes we do use LAPACK for all the LA stuff, and usually LAPACK would come shipped with blas.

File: pytensor/link/c/cmodule.py [hunk 2]
```diff
@@ -2764,6 +2850,11 @@ def default_blas_ldflags():
 
                 # Canopy
         if "Canopy" in sys.prefix:
+            warnings.warn(
```
Comment (twiecki) [inline pytensor/link/c/cmodule.py:2853]: I think we could actually remote those, I don't think they exist. Or just directly raise an Error.

Comment (lucianopaz) [reply pytensor/link/c/cmodule.py:2853]: We should remove them, but I don’t we should do so without a warning that lasts for one minor version

Comment (twiecki) [reply pytensor/link/c/cmodule.py:2885]: I'm pretty sure the % of users using `canopy` and `pytensor` is 0. From https://docs.enthought.com/canopy/index.html: "Canopy provides Python 2.7 and 3.5".

**Comment by lucianopaz** [comment]
I've decided to completely change the `default_blas_ldflags` function. The past implementation was a mess. It tried to import `mkl`, had a bunch of patches here and there to try to work around old conda issues, support something called Canopy and EPD. I decided to simplify the logic a lot, so this might lead to breaks. The idea is the following:

1. Determine the search path of the compiler and linker
2. Use system tools to look at those directories and see if the required BLAS libraries are present or not
3. There's a list of priority libraries that are searched for. 1) MKL with Intel threading, 2) MKL with GNU threading, 3) openblas, 4) LAPACK, 5) plain and simple BLAS
4. If they are present, try to compile a simple program that links with BLAS libraries
    1. If the compilation works handle some OMP stuff and finally return the `ldflags`. At this point I left some old logic about OMP thread handling in place because I didn't want to touch it, but in the future we could refactor it
    2. If the compilation does not work, then move on to the next set libraries to link against
5. If no combination of libraries works, then return `""`, which means that `pytensor` didn't find a blas installation and will use the numpy C-API, which might be slow.

File: pytensor/link/c/cmodule.py [hunk 3]
```diff
@@ -1881,6 +1882,7 @@ def _try_compile_tmp(
             args = cls.compile_args()
         else:
             args = []
+        args = ["--verbose", "-Wl,--verbose"] + args
```
Comment (lucianopaz) [inline pytensor/link/c/cmodule.py:1885]: Oh no, I missed removing this while debugging

**Comment by twiecki** [comment]
Seems like the function just moved to `numpy.distutils.system_info.get_info("blas_opt")`?

**Comment by lucianopaz** [comment]
> Seems like the function just moved to `numpy.distutils.system_info.get_info("blas_opt")`?

The `distutils` module is [deprecated and will be removed](https://numpy.org/doc/stable/reference/distutils.html). Numpy is moving towards using `meson` as its build toolchain. That’s why I had to go with a different more long term solution for this problem

**Comment by twiecki** [comment]
This is with OSX & accelerate:
```
>>ipython
Python 3.11.4 | packaged by conda-forge | (main, Jun 10 2023, 18:08:41) [Clang 15.0.7 ]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.14.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import pytensor as pt
WARNING (pytensor.tensor.blas): Using NumPy C-API based implementation for BLAS functions.

In [2]: pt.__file__
Out[2]: '/Users/twiecki/projects/pytensor/pytensor/__init__.py'

In [3]: pt.config
Out[3]: <pytensor.configparser.PyTensorConfigParser at 0x109603610>

In [4]: pt.config.blas__ldflags
Out[4]: ''
```

**Comment by lucianopaz** [comment]
> This is with OSX & accelerate:
> 
> ```
> >>ipython
> Python 3.11.4 | packaged by conda-forge | (main, Jun 10 2023, 18:08:41) [Clang 15.0.7 ]
> Type 'copyright', 'credits' or 'license' for more information
> IPython 8.14.0 -- An enhanced Interactive Python. Type '?' for help.
> 
> In [1]: import pytensor as pt
> WARNING (pytensor.tensor.blas): Using NumPy C-API based implementation for BLAS functions.
> 
> In [2]: pt.__file__
> Out[2]: '/Users/twiecki/projects/pytensor/pytensor/__init__.py'
> 
> In [3]: pt.config
> Out[3]: <pytensor.configparser.PyTensorConfigParser at 0x109603610>
> 
> In [4]: pt.config.blas__ldflags
> Out[4]: ''
> ```

It’s not surprising that the flags fail to link to accelerate. I could add some platform and processor checks to this to actually only try mkl on Intel and to add the -framework flag on macOS

**Comment by twiecki** [comment]
I added a test for conda:
```python
   # Check if in a Conda/Mamba environment
    conda_prefix = os.environ.get('CONDA_PREFIX')
    if conda_prefix:
        # Check for the existence of typical BLAS libraries in the Conda environment
        blas_libs_path = os.path.join(conda_prefix, 'lib')
        if os.path.exists(os.path.join(blas_libs_path, 'libblas.so')):
            return "-lblas"
        elif os.path.exists(os.path.join(blas_libs_path, 'libopenblas.so')):
            return "-lopenblas"
        elif sys.platform == 'darwin' and os.path.exists(os.path.join(blas_libs_path, 'libblas.dylib')):
            return "-lblas"
        # Add checks for other BLAS libraries as necessary
```

that correctly selects accelerate on my osx.

**Comment by lucianopaz** [comment]
@twiecki, can you try again now? The only difference was that I hadn't included the `dylib` extension into the default searched libraries.

**Comment by twiecki** [comment]
@lucianopaz That didn't help/fix the issue.

**Comment by twiecki** [comment]
It seems we're running into this issue: https://github.com/klee/klee/issues/591

Changing             `extra_compile_flags=[f"-Wl,-rpath={rpath}"] if rpath is not None else [],` to `            extra_compile_flags=[f"-Wl,-rpath,{rpath}"] if rpath is not None else [],` fixes the issue.

File: pytensor/link/c/cmodule.py [hunk 4]
```diff
@@ -2715,203 +2716,139 @@ def default_blas_ldflags():
...
+            extra_compile_flags=[f"-Wl,-rpath={rpath}"] if rpath is not None else [],
+        )
+    except Exception:
+        pass
+    try:
+        # 5. Try to use blas
+        return check_libs(
+            all_libs,
+            required_libs=["blas", "cblas"],
+            extra_compile_flags=[f"-Wl,-rpath={rpath}"] if rpath is not None else [],
```
Comment (twiecki) [inline pytensor/link/c/cmodule.py:2847]:
```suggestion
            extra_compile_flags=[f"-Wl,-rpath,{rpath}"] if rpath is not None else [],
```

**Comment by lucianopaz** [comment]
> (replacing = with a comma).

I didn't know that the `clang` linker complained about the equal sign but `ld` didn't. I pushed the edit and the `rpath` flag should work now

**Comment by twiecki** [comment]
@lucianopaz It now detects blas on OSX with clang. However, it prefers `openblas` over `accelerate` (3 over 5). I recommend to add a check specifically for how conda selects blas, as that covers 90% of the installs, and the subsequent check will work for other OS:
```
   # Check if in a Conda/Mamba environment
    conda_prefix = os.environ.get('CONDA_PREFIX')
    if conda_prefix:
        # Check for the existence of typical BLAS libraries in the Conda environment
        blas_libs_path = os.path.join(conda_prefix, 'lib')
        if os.path.exists(os.path.join(blas_libs_path, 'libblas.so')):
            return "-lblas"
        elif os.path.exists(os.path.join(blas_libs_path, 'libopenblas.so')):
            return "-lopenblas"
        elif sys.platform == 'darwin' and os.path.exists(os.path.join(blas_libs_path, 'libblas.dylib')):
            return "-lblas"
        # Add checks for other BLAS libraries as necessary
```

**Comment by twiecki** [comment]
Alternatively, I'm not sure what `blas` (5) implementation is referring to, can we just move that up?

**Comment by lucianopaz** [comment]
> @lucianopaz It now detects blas on OSX with clang. However, it prefers `openblas` over `accelerate` (3 over 5). I recommend to add a check specifically for how conda selects blas, as that covers 90% of the installs, and the subsequent check will work for other OS:
> 
> ```
>    # Check if in a Conda/Mamba environment
>     conda_prefix = os.environ.get('CONDA_PREFIX')
>     if conda_prefix:
>         # Check for the existence of typical BLAS libraries in the Conda environment
>         blas_libs_path = os.path.join(conda_prefix, 'lib')
>         if os.path.exists(os.path.join(blas_libs_path, 'libblas.so')):
>             return "-lblas"
>         elif os.path.exists(os.path.join(blas_libs_path, 'libopenblas.so')):
>             return "-lopenblas"
>         elif sys.platform == 'darwin' and os.path.exists(os.path.join(blas_libs_path, 'libblas.dylib')):
>             return "-lblas"
>         # Add checks for other BLAS libraries as necessary
> ```

Conda is detected by default. This is done in two ways:

1. if the c compiler being used comes from conda, its search-path will be the environment’s lib directory. Since we look at the compiler’s search path by default, we’ll always find this.
2. Sometimes the system compiler is used (for example in some conda base environments and in non conda installations) so the search path will only include system level directories. What we do there is to add the python sys-lib directory to the search path. In conda installations, that will have the conda lib directory, in pip installations or venvs that will be a directory with some python libs but no blas libraries.

Those two steps cover all conda cases and many other installation modes, so there is no need to add extra checks for conda env vars.

What you say about finding openblas instead of blas by default I think shouldn’t be a problem. Accelerate exposes the blas interface and calls the dylib blas but I don’t think there is a way for us to tell that the dylib is coming from accelerate, openblas or some generic blas implementation. I think that it’s sensible to think that Mac users that have accelerates won’t have openblas or lapack, so those checks will be skipped to finally reach the simple blas case.

**Comment by twiecki** [comment]
But then I'm confused why on OSX with accelerate installed it selects openblas:
```
-L/Users/twiecki/micromamba/envs/pymc5/lib -lopenblas -lgfortran -lgomp -lm -fopenmp -Wl,-rpath,/Users/twiecki/micromamba/envs/pymc5/lib
```

**Comment by twiecki** [comment]
On `main` I'm getting:
```
-L/Users/twiecki/micromamba/envs/pymc5/lib -lcblas -lblas -lcblas -lblas -Wl,-rpath,/Users/twiecki/micromamba/envs/pymc5/lib
```

**Comment by twiecki** [comment]
One other thing I stumbled across is passing the `-framework accelerate` flag (https://stackoverflow.com/questions/66474093/how-do-i-link-the-accelerate-framework-to-a-c-program-in-macos).

**Comment by twiecki** [comment]
I think we can solve this by just moving `openblas` last. You can see here: https://github.com/conda-forge/pytensor-suite-feedstock/pull/71/files that it's really only for non-standard OS. For Intel chips we use MKL for ARM64 we use accelerate. That covers 99%.

**Comment by lucianopaz** [comment]
> One other thing I stumbled across is passing the `-framework accelerate` flag (https://stackoverflow.com/questions/66474093/how-do-i-link-the-accelerate-framework-to-a-c-program-in-macos).

Yeah, I saw that too. The `framework` argument is Mac specific and it works as a shorthand for passing compilation include directories, library dirs and libraries to link against. From the source code of `pytensor`, it looks like we don't need to pass include directories for BLAS because we inline all of the `extern` definitions in the generated source code itself, so we only really need to handle the libraries and shared objects as we are already doing for Linux and gcc.

What I find odd, is that `openblas` seems to make a symbolic link that points `libblas.so` to the `libopenblas.so` library (at least in Ubuntu). Does it do the same for Mac? In that case, openblas would be overwriting whatever comes with Mac's vecLib or accelerate framework.

I'm not sure if it's common practice for the BLAS implementations to alias the `blas` library to link against. If it is, then we could simplify things to use MKL, BLAS+LAPACK or BLAS.

**Comment by twiecki** [comment]
Doesn't look like it:

```bash
>>ls -la /Users/twiecki/micromamba//envs/pymc5/lib/*blas*
lrwxr-xr-x  1 twiecki  staff    22B Jul 11 21:20 /Users/twiecki/micromamba//envs/pymc5/lib/libblas.3.dylib -> libvecLibFort-ng.dylib
lrwxr-xr-x  1 twiecki  staff    22B Jul 11 21:20 /Users/twiecki/micromamba//envs/pymc5/lib/libblas.dylib -> libvecLibFort-ng.dylib
lrwxr-xr-x  1 twiecki  staff    22B Jul 11 21:20 /Users/twiecki/micromamba//envs/pymc5/lib/libcblas.3.dylib -> libvecLibFort-ng.dylib
lrwxr-xr-x  1 twiecki  staff    22B Jul 11 21:20 /Users/twiecki/micromamba//envs/pymc5/lib/libcblas.dylib -> libvecLibFort-ng.dylib
-rwxrwxr-x  3 twiecki  staff   9.9M Apr 13  2023 /Users/twiecki/micromamba//envs/pymc5/lib/libopenblas.0.dylib
lrwxr-xr-x  1 twiecki  staff    29B Jul  7 23:51 /Users/twiecki/micromamba//envs/pymc5/lib/libopenblas.a -> libopenblas_vortexp-r0.3.23.a
lrwxr-xr-x  1 twiecki  staff    19B Jul  7 23:51 /Users/twiecki/micromamba//envs/pymc5/lib/libopenblas.dylib -> libopenblas.0.dylib
lrwxr-xr-x  1 twiecki  staff    19B Jul  7 23:51 /Users/twiecki/micromamba//envs/pymc5/lib/libopenblas_armv8p-r0.3.23.dylib -> libopenblas.0.dylib
-rw-rw-r--  3 twiecki  staff    17M Apr 13  2023 /Users/twiecki/micromamba//envs/pymc5/lib/libopenblas_vortexp-r0.3.23.a
lrwxr-xr-x  1 twiecki  staff    19B Jul  7 23:51 /Users/twiecki/micromamba//envs/pymc5/lib/libopenblas_vortexp-r0.3.23.dylib -> libopenblas.0.dylib
lrwxr-xr-x  1 twiecki  staff    19B Jul  7 23:51 /Users/twiecki/micromamba//envs/pymc5/lib/libopenblasp-r0.3.23.dylib -> libopenblas.0.dylib
```

**Comment by lucianopaz** [comment]
Ok, so I'll put BLAS ahead of openblas, but I'm not sure if that will ever get used as a default

**Comment by twiecki** [comment]
> Ok, so I'll put BLAS ahead of openblas, but I'm not sure if that will ever get used as a default

It will on OSX:

```
>>python -c "import pytensor; print(pytensor.config.blas__ldflags)"
-L/Users/twiecki/micromamba/envs/pymc5/lib -llapack -lblas -lcblas -lm -Wl,-rpath,/Users/twiecki/micromamba/envs/pymc5/lib
```

**Review by twiecki — APPROVED** [review]

**Comment by twiecki** [comment]
Have we tested on windows?

**Comment by lucianopaz** [comment]
> > Ok, so I'll put BLAS ahead of openblas, but I'm not sure if that will ever get used as a default
> 
> It will on OSX:
> 
> ```
> >>python -c "import pytensor; print(pytensor.config.blas__ldflags)"
> -L/Users/twiecki/micromamba/envs/pymc5/lib -llapack -lblas -lcblas -lm -Wl,-rpath,/Users/twiecki/micromamba/envs/pymc5/lib
> ```

Sorry, I meant to say that I don't think that openblas will ever end up being used as a default.

**Comment by twiecki** [comment]
> Sorry, I meant to say that I don't think that openblas will ever end up being used as a default.

Yes, that's what we want and what I was saying here:

> You can see here: https://github.com/conda-forge/pytensor-suite-feedstock/pull/71/files that it's really only for non-standard OS. For Intel chips we use MKL for ARM64 we use accelerate. That covers 99%.

**Comment by lucianopaz** [comment]
> Have we tested on windows?

I’ll test it

**Comment by twiecki** [comment]
> Do you need me to test nothing is changed for linux amd64 users (like me)?

Sure, the more testing the better. This one could cause quite some disruption if we get it wrong.

**Comment by ricardoV94** [comment]
This is what I got before:
```
-L/home/ricardo/miniconda3/envs/pytensor/lib -lcblas -lblas -lcblas -lblas -Wl,-rpath,/home/ricardo/miniconda3/envs/pytensor/lib
```
And what I get after:
```
-L/usr/lib/x86_64-linux-gnu -L/home/ricardo/miniconda3/envs/pytensor/lib -L/usr/lib/gcc/x86_64-linux-gnu/9 -llapack -lblas -lcblas -lm -Wl,-rpath,/home/ricardo/miniconda3/envs/pytensor/lib
```

**Comment by twiecki** [comment]
Lgtm.

**Comment by twiecki** [comment]
Actually, I don't think `-L/usr/lib/x86_64-linux-gnu` should show up there as it might override the env settings.

**Comment by lucianopaz** [comment]
> Actually, I don't think `-L/usr/lib/x86_64-linux-gnu` should show up there as it might override the env settings.

That directory came from the compiler default link directories. That means that it is consistent with the env settings and the compiler configuration. Maybe the safest thing is to not include `-L` flags to folders that are already part of the compiler or linker’s search path, because they might affect the lookup order when linking.

**Comment by twiecki** [comment]
@lucianopaz Any thoughts on this? Should we just merge? CC @ricardoV94

**Comment by lucianopaz** [comment]
> @lucianopaz Any thoughts on this? Should we just merge? CC @ricardoV94

I'll remove the compiler libs from the -L flags and then I think we can merge

**Review by twiecki — APPROVED** [review]

**Comment by lucianopaz** [comment]
@ricardoV94, do you want to try this out on your AMD setup and see if the link flags work for you there?

**Comment by ricardoV94** [comment]
> @ricardoV94, do you want to try this out on your AMD setup and see if the link flags work for you there?

This is what I get now

```
-L/home/ricardo/miniconda3/envs/pytensor/lib -llapack -lblas -lcblas -lm -Wl,-rpath,/home/ricardo/miniconda3/envs/pytensor/lib
```

**Comment by twiecki** [comment]
@ricardoV94 that looks perfect to me

**Comment by lucianopaz** [comment]
@ricardoV94, LGTM

**Comment by twiecki** [comment]
Congrats on slaying this dragon @lucianopaz!

Time for a new release.

**Pull request merged**